appveyor.yml

```diff
@@ -1,23 +1,15 @@
 version: 4.0.0.{build}
 
-os: Visual Studio 2019
-
 branches:
   except:
     - configdata
     - gh-pages
 
-environment:
-  # The following entries define the default values of the environment variables.
-  # They can be overridden in AppVeyor | Settings | Environment.
-  matrix:
-  - IdeVersion: VS2019
-  SKIP_PAUSE: TRUE
-  ARCHIVE_WITH_PDB: TRUE
-  GE_TEST_SLEEP_SECONDS_ON_HANG: 0
-  GE_TEST_LAUNCH_DEBUGGER_ON_HANG: 0
-  TEST_RUN_COUNT: 1
-  IGNORE_TEST_RESULT: FALSE
```

Comment:
Why did you remove all AppVeyor variables?
These four still make sense to me.

Reply:
They are useful for debug builds, but we shouldn't have them in the main build.
The master and the release builds aren't meant to be used as test grounds. For tests we can create branches off the main trunk and modify appveyor.yml as required to complete tests.

Reply:
Git does not support this in a user-friendly way. This will end up in lots of merges, rebases and merge conflicts.
I don't need it as AppVeyor/Environment variables -- I was told to use them.
But the "extended test infrastructure" should be available and maintained in the main trunk.

BTW: The build script seems to *not* set the exit code.

Reply:
> I don't need it as AppVeyor/Environment variables -- I was told to use them.

I suggested to use AppVeyor env variables as a way to dynamically change behaviours, if that's what you're referring to. It is certainly a better way than hardcoding magic numbers in the common infrastructure.

> But the "extended test infrastructure" should be available and maintained in the main trunk.

Why?
The master and release branches must be stable. If we have issues with tests (like now), we should create a feature branch in the main trunk and tweak its pipeline to suit (e.g. run tests multiple times, ignore errors, etc). Once the work is complete, the fix is submitted to the master, and all temporary tweaks are discarded.

Reply:
> BTW: The build script seems to _not_ set the exit code.

Could you please elaborate?

In a typical flow, `Build` function is invoked and returns the code:
https://github.com/gitextensions/gitextensions/blob/c110ab60f1ab3075b3e8e8213a1990da698fd006/Build/Build.ps1#L75-L91

If there is an exception, it returns here:
https://github.com/gitextensions/gitextensions/blob/c110ab60f1ab3075b3e8e8213a1990da698fd006/Build/Build.ps1#L113-L116

What am I missing?

Reply:
> Once the work is complete, the fix is submitted to the master, and all temporary tweaks are discarded.

"discard" is the point: I do not want to add the loop around the tests over and over again.
I just want to temporarily set `$testRunCount` from `1` to a suiting value in `AppVeyor.yml` or in a build script of such a feature branch.
AppVeyor variables do not work well because they are global.

> BTW: The build script seems to not set the exit code.

I guess it has not worked before the one week old c110ab60f1ab3075b3e8e8213a1990da698fd006, which I did not take notice of.
And `$host.SetShouldExit` seems to not exit at once, which confused me.

Reply:
> "discard" is the point: I do not want to add the loop around the tests over and over again.
> I just want to temporarily set `$testRunCount` from `1` to a suiting value in `AppVeyor.yml` or in a build script of such a feature branch.
> AppVeyor variables do not work well because they are global.

This comes down to the point I'm trying to make - master or release/* branches are not the right place to test things.
We can have an `experimental` or `test` branch off the main trunk where appveyor.yml can be modified to whatever state maybe necessary, and have env variables that dynamically alter its behaviours.

Reply:
We can create a separate build project to build a specific branch (e.g. `experimental`) that would use the custom yml file.

Reply:
I am not convinced. It _will_ get out of sync. But go ahead. If we are lucky, we won't need it anymore.